src/field_theory/ratfunc.lean

```diff
@@ -52,6 +52,12 @@ namely `ratfunc.of_fraction_ring`, `ratfunc.to_fraction_ring`, `ratfunc.mk` and
 All these maps get `simp`ed to bundled morphisms like `algebra_map (polynomial K) (ratfunc K)`
 and `is_localization.alg_equiv`.
 
+## References
+
+* [Kleiman, *Misconceptions about $K_X$*][kleiman1979]
+* https://freedommathdance.blogspot.com/2012/11/misconceptions-about-kx.html
```

Comment:
I didn't know about these, and I'm sure I would have fallen in these traps if I came close to them. Thanks!

But are they directly relevant to what is happening in this PR? Or do you include them as future warning?

Reply:
The `rafunc.lift_on`, as it was designed by Anne, was defined to require that the commutative ring `K` is an integral domain. I was working on lifting `polynomial.taylor` into something that is similar for `ratfunc` and since `taylor` does not require `no_zero_divisors`, lifting the function made sense without requiring `is_domain`. I could have used `localization.lift_on` directly, but that seemed to break the API layer.

I asked Kevin whether anyone care about defining functions/transformations on rational functions `K(x)` where `K` is a commutative ring that is *not* an integral domain. He said, yes, here is a reference that talks about fraction rings where zero divisors exist in the base ring.

I am not defining any such complex functions or algebraic structures here, just including the references as motivation for why `ratfunc.lift_on` need not require `[is_domain K]`. We still have Anne's `lift_on'` which is restricted to integral domains, and has a somewhat simpler lifting condition.

Please let me know if these thoughts should be included in the code anywhere.

Reply:
Thanks for explaining the context